Add `/topics` pages (#644)
OpenSats/website: data/topics/private-dms.mdx

---
title: 'Private DMs'
summary: 'The current standard for private direct messages on Nostr, using gift wrap to hide sender, receiver, and metadata from relays.'
category: 'Nostr'
aliases: ['NIP-17', 'NIP-59', 'gift wrap', 'Nostr DMs']
---

> NIP-17 is the current standard for private direct messages on Nostr. It replaces the older NIP-04, which leaked both the sender and the receiver of every message to any relay that stored it. Under [NIP-17](/topics/private-dms), messages are encrypted with NIP-44 and wrapped in the "gift wrap" construction from [NIP-59](/topics/private-dms), which also hides the real timestamp and other metadata.

Reviewer: ```suggestion
NIP-17 is the current standard for private direct messages on [Nostr](/topics/nostr). It replaces the older NIP-04, which leaked both the sender and the receiver of every message to any relay that stored it. Under [NIP-17](/topics/private-dms), messages are encrypted with NIP-44 and wrapped in the "gift wrap" construction from [NIP-59](/topics/private-dms), which also hides the real timestamp and other metadata.
```

Reviewer: Also auto-references

> Each DM ends up stored on relays as an event whose apparent sender and receiver look random. Only the real recipient can unwrap it, and only after doing so do they learn who sent it. Relays cannot correlate conversations or build a social graph from DM traffic.

Reviewer: ```suggestion
Each DM ends up stored on relays as an event whose apparent sender and receiver look random. Only the real recipient can unwrap it, and only after doing so he is able to learn who sent it. Relays cannot correlate conversations or build a social graph from DM traffic.
```
